src/products/models/dtos/product.dto.ts

```diff
@@ -0,0 +1,43 @@
+import { CategoryProducts } from '@category_products/mc/category_products.entity';
+import { IsArray, IsInt, IsOptional, IsString } from 'class-validator';
+import { Product } from '../classes/product.entity';
+
+export class ProductDTO {
+  @IsString()
+  name: string;
+
+  @IsString()
+  description: string;
+
+  @IsInt()
+  price: number;
+
+  @IsInt()
+  rating: number;
+
+  @IsInt()
+  status: number;
+
+  @IsInt()
+  stock: number;
+
+  @IsArray()
+  @IsOptional()
+  categories: number[];
+
+  toProduct(): Product {
```

Comment:
DTO Responsibility is validate the data. If you want to create an entity or an object for a specific class, you can use the class that you define in models/classes and use a constructor to build objects.

Reply:
I moved it to the Product's constructor, although I realized you can only have 1 constructor per class so this might be cumbersome when scaling